src/engine/QueryPlanner.cpp

```diff
@@ -554,7 +616,15 @@ bool QueryPlanner::checkUsePatternTrick(
                   break;
                 }
               }
-
+            } else if constexpr (std::is_same_v<T,
+                                                GraphPatternOperation::Bind>) {
+              // check if pattern trick object has been used in Bind
```

Comment:
// If the object variable of ql:has-predicate is used somewhere in a BIND, we cannot use the pattern trick.

PS: Why do we need this code block twice?

Reply:
I am planning on changes, that make this whole mess go away.
It is indeed full duplication. The explanation is not worth the time.
Previously much programming in QLever has been done by copying and pasting large portions of code. I already have begun mitigating this.

Reply:
OK, thanks for the explanation + agreed that we should leave it as it is in this PR